README.md

```diff
@@ -1,4 +1,231 @@
+# acf-field-group-composer
+
+[![standard-readme compliant](https://img.shields.io/badge/readme%20style-standard-brightgreen.svg?style=flat-square)](https://github.com/RichardLitt/standard-readme)
 [![Build Status](https://img.shields.io/travis/bleech/acf-field-group-composer.svg?style=flat-square)](https://travis-ci.org/bleech/acf-field-group-composer)
-<!-- [![Style Check](https://styleci.io/repos/73226024/shield)](https://styleci.io/repos/73226024) -->
 [![Code Quality](https://img.shields.io/scrutinizer/g/bleech/acf-field-group-composer.svg?style=flat-square)](https://scrutinizer-ci.com/g/bleech/acf-field-group-composer)
 [![Code Coverage](https://img.shields.io/coveralls/bleech/acf-field-group-composer.svg?style=flat-square)](https://coveralls.io/github/bleech/acf-field-group-composer)
+
+> Configuration builder for advanced custom fields field groups
+
+This plugin helps you create reusable ACF fields and field groups with code.
+
+## Table of Contents
+
+- [Background](#background)
+- [Install](#install)
+- [Usage](#usage)
+- [API](#api)
+- [Maintainers](#maintainers)
+- [Contribute](#contribute)
+- [License](#license)
+
+## Background
+
+ACF provides a great interface to create custom meta boxes in WordPress. However, there are to major downsides using it out of the box.
+
+First, configuring it via its GUI can be cumbersome if you have a lot of custom fields. Also fields created via the GUI are only stored in the database. That means migrating them between different stages of a development setup can only be done by migrating the database as well.
+
+Second, the local JSON feature, or `add_local_field_group`, is handy for checking the config into SCM and deploying it to different stages but it lacks some customizability, like reusing already specified fields in different context.
+
+acf-field-group-composer helps circumventing these downsides. It let's you use the build in `add_local_field_group` function, automatically adds unique keys to all fields added, and, if a field is not an array, but a string, it will apply a filter with that name and use the return value as a new field in the config.
```

Comment:
"It let's you use the **built-in** `add_local_field_group` function, **and** automatically adds unique keys to all fields added. **Plus, if** a field is not an array, but a string, it will apply a filter with that name and use the return value as a new field in the config."